gcloud-java-dns/src/test/java/com/google/gcloud/dns/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,1405 @@
+    assertTrue(RPC.deleteZone(ZONE1.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    // deleting non-existent zone
+    assertFalse(RPC.deleteZone(ZONE1.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    RPC.create(ZONE2, EMPTY_RPC_OPTIONS);
+    assertNotNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    assertNotNull(RPC.getZone(ZONE2.getName(), EMPTY_RPC_OPTIONS));
+    // delete in reverse order
+    assertTrue(RPC.deleteZone(ZONE1.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    assertNotNull(RPC.getZone(ZONE2.getName(), EMPTY_RPC_OPTIONS));
+    assertTrue(RPC.deleteZone(ZONE2.getName()));
+    assertNull(RPC.getZone(ZONE1.getName(), EMPTY_RPC_OPTIONS));
+    assertNull(RPC.getZone(ZONE2.getName(), EMPTY_RPC_OPTIONS));
+    RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    RPC.applyChangeRequest(ZONE1.getName(), CHANGE_KEEP, EMPTY_RPC_OPTIONS);
+    try {
+      assertFalse(RPC.deleteZone(ZONE1.getName()));
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(400, ex.code());
+    }
+  }
+
+  @Test
+  public void testCreateAndApplyChangeWithThreads() {
+    LocalDnsHelper localDnsThreaded = LocalDnsHelper.create(1000L); // using threads here
+    localDnsThreaded.createZone(PROJECT_ID1, ZONE1, null);
+    assertNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("1"));
+    LocalDnsHelper.Response response =
+        localDnsThreaded.createChange(PROJECT_ID1, ZONE_NAME1, CHANGE1, null); // add
+    assertEquals(200, response.code());
+    assertNotNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("1"));
+    assertNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("2"));
+    localDnsThreaded.createChange(PROJECT_ID1, ZONE_NAME1, CHANGE1, null); // add
+    response = localDnsThreaded.createChange(PROJECT_ID1, ZONE_NAME1, CHANGE1, null); // add
+    assertEquals(200, response.code());
+    assertNotNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("1"));
+    assertNotNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("2"));
+    localDnsThreaded.createChange(PROJECT_ID1, ZONE_NAME1, CHANGE2, null); // delete
+    assertNotNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("1"));
+    assertNotNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("2"));
+    assertNotNull(localDnsThreaded.findZone(PROJECT_ID1, ZONE_NAME1).findChange("3"));
+    localDnsThreaded.createChange(PROJECT_ID1, ZONE_NAME1, CHANGE_KEEP, null); // id is "4"
+    // check execution
+    Change change = localDnsThreaded.findChange(PROJECT_ID1, ZONE_NAME1, "4");
+    for (int i = 0; i < 10 && !change.getStatus().equals("done"); i++) {
+      // change has not been finished yet; wait at most 20 seconds
+      // it takes 5 seconds for the thread to kick in in the first place
```

Comment:
1 second is better than the original of 5 seconds, but do you think this could be changed to something like 50 ms?  Also update "5 seconds" to whatever we end up choosing.